gemini/sample-apps/finance-advisor-spanner/pages/1_Asset_Search.py

```diff
@@ -0,0 +1,83 @@
+"""This module is the page for Asset Search feature"""
+
+# pylint: disable=line-too-long,import-error
+
+from database import display_spanner_query, fts_query, like_query
+from home import table_columns_layout_setup
+from itables.streamlit import interactive_table
+import streamlit as st
+
+st.logo(
+    "https://storage.googleapis.com/github-repo/generative-ai/sample-apps/finance-advisor-spanner/images/investments.png"
+)
+
+
+def asset_search_precise() -> None:
+    """This function implements Asset search LIKE Query"""
+
+    query_params = []
+    query_params.append(investment_strategy_pt1.strip())
+    query_params.append(and_or_exclude)
+    query_params.append(investment_strategy_pt2.strip())
+    query_params.append(investment_manager.strip())
+    asset_search_common(query_params, "PRECISE")
+
+
+def asset_search_common(query_params: list, query_type: str) -> None:
+    """This function implements Asset search common  functions"""
+
+    st.header("FinVest Fund Advisor")
+    st.subheader("Asset Search")
+
+    with st.spinner("Querying Spanner..."):
+        if query_type == "PRECISE":
+            return_vals = like_query(query_params)
+        else:
+            return_vals = fts_query(query_params)
+        spanner_query = return_vals.get("query")
+        data = return_vals.get("data")
+        display_spanner_query(str(spanner_query))
+
+    interactive_table(data, caption="", **table_columns_layout_setup())
+
+
+def asset_search() -> None:
+    """This function implements Asset Search"""
+    query_params = []
+    query_params.append(investment_strategy)
+    query_params.append(investment_manager)
+    asset_search_common(query_params, "FTS")
```

Comment:
Does this need a dedicated method? It's only called one time, seems like this could just be added to the place it's called from like:

```py
asset_search_common([investment_strategy, investment_manager], "FTS")
```

Reply:
fixed